### DISCUSSION TITLE
Question regarding TSR, power output and the state of axial induction

### DISCUSSION THREAD

**Discussion by MarcusBinder** [NatLabRockies/floris#353] [Q&A] [answered]
Hello

**In short**: I have some questions regarding how the power output for the turbines are calculated, and how the turbines TSR comes into play. Also I am unsure about the implementation of axial induction.

**Slightly longer**: I am trying to learn floris in preparation for my master thesis, and for that I have come upon a couple of questions. 
I am now trying to add a Vestas V80 turbine instead of the inbuilt NREL5MW that comes with the examples, but I can see that one of the input parameters is the TSR. I am still new to this, so I don’t understand what this parameter does.
Also is it correctly understood that the power output is calculated independent of the axial induction? 

I hope that my questions make sense. Thanks in advance! 

Also im using floris 3.0 if that makes in difference

**Comment by Bartdoekemeijer** [comment, marked as answer]
Hi Marcus,

First, a turbine is typically controlled differently at different wind speed regions. You may already be familiar with this, but repeating it here for completeness.

![image](https://user-images.githubusercontent.com/22119448/156420352-1d998470-3a0c-4e94-b3e0-14c830e33131.png)
Pao and Johnson, "A Tutorial On The Dynamics And Control Of Wind Turbines And Wind Farms", 2009, American Control Conference Proceedings

Secondly, the aeroelastic curves of a turbine are typically defined by tip-speed-ratio (TSR) and the blade pitch angle. Typically, if a turbine wants to maximize power capture (region II), it will do its best to stay on the top of this curve (maximum cp) by controlling the rotor speed (i.e., TSR) and blade pitch angles. Towards region III, the turbine starts producing its rated power and the turbine will curtail and move to a lower cp on the curve. There are different ways of derating a turbine, as nicely described and illustrated in Van Der Hoek, et al.:

![image](https://user-images.githubusercontent.com/22119448/156419496-dd3741f9-3f94-4ef2-9cfe-3114afe5ac9c.png)
Van Der Hoek et al., _Comparison of Down-Regulation Strategies for Wind Farm Control and their Effects on Fatigue Loads_, 2018, American Control Conference Proceedings

Basically, different wind turbine controllers will decide to move differently over the Cp curve for the wind speed range. This means that different controllers will impact the turbine's thrust and power production differently. It is complicated and not desirable to capture such detail in an engineering wind farm flow model like FLORIS.

To circumvent the need to implement an entire wind turbine control algorithm inside FLORIS, we abstractify the turbine's behavior with a table of wind speed against Cp (normalized power) and Ct (normalized thrust). One can achieve such a table by implementing a your turbine controller of interest in OpenFAST or another aeroelastic turbine simulator for the full range of wind speeds and extracting the power and thrust values. We could also consider adding a dimension to this table like a power setpoint: e.g., extract 100% of the power available, 90% of the power available, ..., 10% of the power available. Since there are multiple ways to curtail a turbine, one should then generate the table for FLORIS by implementing their method in an actual turbine control algorithm and simulating all relevant conditions in an aeroelastic turbine simulator like OpenFAST. Actually, Issue #314 will address this.

What typically happens if we dont have a full model of the turbine available for simulations with OpenFAST, is that we simply scale the NREL 5MW `cp` and `ct` curves to the turbine of interest. Basically, we scale it such that the cut-in (region I) and rated wind speed align for the turbine of interest, and that at region III the turbine captures exactly the rated power.

I hope this clarifies things. Let me know if you have additional questions.

Best,
Bart

**Reply by MarcusBinder** [reply]
Hi Bart

Thanks for the fast reply!  That makes a ton of sense. 

Is it correctly understood then that the TSR value is only used in the program to give a reference for the Cp and Ct values and not used directly in the power output calculation? 

Thanks again, 
Marcus

**Reply by Bartdoekemeijer** [reply]
Hi Marcus,

You make a great point. Searching through the repository, it seems that `TSR` is not actually used anywhere. @bayc Would you by any chance know the reason for having `TSR` specified in the turbine definitions? They are passed around inside the `floris` object but not used in any calculation, as far as I can see.

**Reply by rafmudaf** [reply]
Tip speed ratio is used in a couple of places:
- [Wake added yaw](https://github.com/NREL/floris/blob/main/floris/simulation/solver.py#L116)
- [V and W](https://github.com/NREL/floris/blob/main/floris/simulation/solver.py#L144)

These go into the wake calculation so they affect the velocity at downstream turbines.

**Reply by Bartdoekemeijer** [reply]
@rafmudaf Ah, you're completely correct!

https://github.com/NREL/floris/blob/4d7733c7ff35e89728a9134222b07d74b836a055/floris/simulation/wake_deflection/gauss.py#L302

Thanks!

**Reply by MarcusBinder** [reply]
Thanks a lot!

**Reply by kaizhang3** [reply]
Hi, reading through, I understand that for each wind velocity, floris looks up the power and thrust coefficients through the Cp-Ct-velocity table. Does this also mean that the TRUE TSR is also implicitly determined by the wind velocity? Since for each control region, the turbine definition also determines how fast the turbine should rotate (see e.g. figure 9-1 of the NREL 5 MW definite at [here](https://www.nrel.gov/docs/fy09osti/38060.pdf)).
If the above is true, then shouldn't the TSR value be replaced by the TRUE TSR when it is used, rather than the single fixed TSR defined in the turbine model ymal file?
